packages/transformer/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -1356,6 +1360,81 @@ describe("IModelTransformer", () => {
     targetIModelDb.close();
   });
 
+  it("should log unresolved references", async () => {
+    const iModelShared: SnapshotDb =
+      IModelTransformerTestUtils.createSharedIModel(outputDir, ["A", "B"]);
+    const iModelA: SnapshotDb = IModelTransformerTestUtils.createTeamIModel(
+      outputDir,
+      "A",
+      Point3d.create(0, 0, 0),
+      ColorDef.green
+    );
+    IModelTransformerTestUtils.assertTeamIModelContents(iModelA, "A");
+    const iModelExporterA = new IModelExporter(iModelA);
+
+    // Exclude element
+    const excludedId = iModelA.elements.queryElementIdByCode(
+      Subject.createCode(iModelA, IModel.rootSubjectId, "Context")
+    );
+    assert.isDefined(excludedId);
+    iModelExporterA.excludeElement(excludedId!);
+
+    const subjectId: Id64String = IModelTransformerTestUtils.querySubjectId(
+      iModelShared,
+      "A"
+    );
+    const transformerA2S = new IModelTransformer(
+      iModelExporterA,
+      iModelShared,
+      { targetScopeElementId: subjectId, danglingReferencesBehavior: "ignore" }
+    );
+    transformerA2S.context.remapElement(IModel.rootSubjectId, subjectId);
+
+    // Configure logger to capture warning message about unresolved references
+    const messageStart = "The following elements were never fully resolved:\n";
+    const messageEnd =
+      "\nThis indicates that either some references were excluded from the transformation\nor the source has dangling references.";
+
+    let unresolvedElementMessage: string | undefined;
+    const logWarning = (
+      _category: string,
+      message: string,
+      _metaData: LoggingMetaData
+    ) => {
+      if (message.startsWith(messageStart)) {
+        unresolvedElementMessage = message;
+      }
+    };
+    Logger.initialize(undefined, logWarning);
+    Logger.setLevelDefault(LogLevel.Warning);
+
+    // Act
+    await transformerA2S.processAll();
+
+    // Collect expected ids
+    const result = iModelA.queryEntityIds({
+      from: "BisCore.Element",
+      where: "Model.Id = :rootId AND ECInstanceId NOT IN (:excludedId)",
```

Comment:
Note I slightly modified this query... previously it said ECINstanceId NOT IN (:rootId, :excludedId), but the rootId started showing up in the actual ids. I assume this was caused by no longer ignoring rootsubject during transformations.